spec/integration/collection_association_spec.rb

```diff
@@ -42,6 +42,22 @@ class Book < ActiveFedora::Base
     end
   end
 
+  describe "#delete" do
+    context "when given items not in collection" do
+      it "should return an empty set" do
+        expect(library.books.delete(Book.new)).to eq []
+      end
+      it "should not act on it" do
+        b = Book.new
+        allow(b).to receive(:persisted?).and_return(true)
+
+        library.books.delete(b)
+
+        expect(b).not_to have_received(:persisted?)
```

Comment:
I thought the have_received syntax was geared towards spies vs. vanilla mocks. Is b a spy here?